### PR TITLE
Add System.IO.FileSystem.Watcher Source and Tests

### PR DESCRIPTION
**Pull request by chcosta**

### CODE REVIEW

File: src/System.IO.FileSystem.Watcher/src/System.IO.FileSystem.Watcher.csproj [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    <Compile Include="System\IO\RenamedEventHandler.cs" />
+    <Compile Include="System\IO\WatcherChangeTypes.cs" />
+  </ItemGroup>
+  <ItemGroup Condition=" '$(OS)' == 'Windows_NT' ">
+    <Compile Include="Microsoft\Win32\NativeMethods.cs" />
+    <Compile Include="Microsoft\Win32\SafeNativeMethods.cs" />
+    <Compile Include="Microsoft\Win32\UnsafeNativeMethods.cs" />
+    <Compile Include="System\IO\FileSystemWatcher.Win32.cs" />
+  </ItemGroup>
+  <ItemGroup Condition=" '$(OS)' == 'Unix' ">
```
Comment (stephentoub) [inline src/System.IO.FileSystem.Watcher/src/System.IO.FileSystem.Watcher.csproj]: The interop files were restructured as part of #531, so this section will need to be updated as well.  This is why the CI build failed, it couldn't find these files.

File: src/System.IO.FileSystem.Watcher/src/Interop/Interop.Linux.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+partial class Interop
+{
+    const string LIBC = "libc";
```
Comment (stephentoub) [inline src/System.IO.FileSystem.Watcher/src/Interop/Interop.Linux.cs]: This file should be renamed and restructured related to #531.  
- It should be in src\Common\src\Interop\Linux\libc\Interop.inotify.cs
- The LIBC constant should be removed.  
- "partial class Interop" should be changed to "internal static partial class Interop"
- Its contents should all be moved to a nested type "internal static partial class libc".

**Comment by nguerrera** [comment]
@chcosta I did not review the commit, but please rewrite away the reverse pull requests before merging this. They are going to create clutter and I'd rather folks learn how to avoid them then take them in to the master history. Grab me when everything else is signed off and I'll help you do it...

In the future to avoid this altogether:
1. Don't work directly in master, always create a topic branch and send PRs from that.
2. Update your fork's master using the instructions at https://help.github.com/articles/syncing-a-fork/, not using a pull request from dotnet -> fork

**Comment by ellismg** [comment]
Should this be against v1.0 since we haven't pulled the trigger on the new branching plan yet?

**Pull request closed without merging**